all/test/all_test.go

```diff
@@ -552,6 +552,40 @@ func (s *TestSuite) TestAllCases() {
 			},
 			extraArgs: []string{"--with-validator"},
 		},
+		"typescript": {
+			lang: "typescript",
+			protofileName: "all/test/test.proto",
+			expectedOutputDir: "gen/pb-typescript",
+			fileExpectations: []FileExpectation{
+				{fileName: "all/test/test.ts"},
```

Comment:
so the typescript plugin doesn't generate the other files that you added to the tests originally?

Reply:
No, those were actually copied from previous tests, and I should have realized something was off when the test passed in the first place because it wasn't what I was originally expecting, but I assumed that the project had some organizational components that I didn't fully understand.

This is the full actual file output from running the test. I was really only expecting `test.ts` -- a file of the same name as the input `.proto` file with the `.ts` extention.

```
typescript095321988/gen/
└── pb-typescript
    ├── all
    │   └── test
    │       ├── include.ts
    │       └── test.ts
    ├── google
    │   ├── api
    │   │   ├── annotations.ts
    │   │   └── http.ts
    │   └── protobuf
    │       ├── descriptor.ts
    │       ├── duration.ts
    │       ├── field_mask.ts
    │       └── timestamp.ts
    └── validate
        └── validate.ts
```

Reply:
is it expected for the ts plugin to generate the dependencies?

Reply:
To resolve the imports in the .proto file, yes: https://github.com/namely/docker-protoc/blob/master/all/test/test.proto#L6

Whether or not this is desirable behavior in itself is a question for ts-proto, but my understanding is that there is no way around it since there is no actual typescript source from Google to rely on, And even if there were it likely wouldn't match all the potential style changes supported by ts-proto arguments (such as useOptionals)

Reply:
that makes sense, thank you.